packages/statemanager/src/stateManager.ts

```diff
@@ -137,10 +152,36 @@ export class DefaultStateManager implements StateManager {
   _storageTries: { [key: string]: Trie }
   _codeCache: { [key: string]: Uint8Array }
 
+  // EIP-2929 address/storage trackers.
+  // This maps both the accessed accounts and the accessed storage slots.
+  // It is a Map(Address => StorageSlots)
+  // It is possible that the storage slots set is empty. This means that the address is warm.
+  // It is not possible to have an accessed storage slot on a cold address (which is why this structure works)
+  // Each call level tracks their access themselves.
+  // In case of a commit, copy everything if the value does not exist, to the level above
+  // In case of a revert, discard any warm slots.
+  //
+  // TODO: Switch to diff based version similar to _touchedStack
+  // (_accessStorage representing the actual state, separate _accessedStorageStack dictionary
+  // tracking the access diffs per commit)
+  protected _accessedStorage: Map<string, Set<string>>[]
```

Comment:
How much of (if any) this caching/journaling stuff is duplicating the various bits of caching that has built out in the recent caching work merged into v7?  I only ask because it feels like we're duplicating a lot of work here (though I do recognize that the `stateManager` is doing different work with the caching it does).  What I'm wondering if is we need a generalized caching mechanism in the `stateManager` when we are tracking the state changes in each call frame as a checkpoint through the journaling here so seems like there's an opportunity to streamline caching.

Reply:
Yes I agree we should check if we can generalize it. Note that the PR is still super WIP. I am now at the phase where I can really start refactoring out things from the old statemanager.

Reply:
Sounds good! Was just giving this an early look

Reply:
Yup, great and thanks :smile: :+1: